custom/ewsghana/reports/specific_reports/stock_status_report.py

```diff
@@ -382,6 +339,156 @@ class StockStatus(MultiReport):
     is_exportable = True
 
     @property
+    def get_supply_points(self):
+        supply_points = []
+        if self.report_config['location_id']:
+            location = SQLLocation.objects.get(
+                domain=self.report_config['domain'],
+                location_id=self.report_config['location_id']
+            )
+            if location.location_type.name == 'country':
+                supply_points = SQLLocation.objects.filter(
+                    Q(parent__location_id=self.report_config['location_id'], is_archived=False) |
+                    Q(location_type__name='Regional Medical Store', domain=self.report_config['domain']) |
+                    Q(location_type__name='Teaching Hospital', domain=self.report_config['domain'])
+                ).order_by('name').exclude(supply_point_id__isnull=True)
+            else:
+                supply_points = SQLLocation.objects.filter(
+                    parent__location_id=self.report_config['location_id'], is_archived=False,
+                    location_type__administrative=False,
+                ).order_by('name').exclude(supply_point_id__isnull=True)
+        return supply_points
+
+    def unique_products(self, locations, all=False):
+        if self.report_config['products'] and not all:
+            return SQLProduct.objects.filter(
+                pk__in=locations.values_list('_products', flat=True),
+            ).filter(pk__in=self.report_config['products']).exclude(is_archived=True)
+        elif self.report_config['program'] and not all:
+            return SQLProduct.objects.filter(
+                pk__in=locations.values_list('_products', flat=True),
+                program_id=self.report_config['program']
+            ).exclude(is_archived=True)
+        else:
+            return SQLProduct.objects.filter(
+                pk__in=locations.values_list('_products', flat=True)
+            ).exclude(is_archived=True)
```

Comment:
@snopoke In stock status report we fetch only all products so here is the query:
```
SELECT ...
      FROM "products_sqlproduct"
      WHERE ("products_sqlproduct"."id" IN
               (SELECT ...
                FROM "locations_sqllocation" U0
                LEFT OUTER JOIN "locations_sqllocation__products" U1 ON (U0."id" = U1."sqllocation_id")
                WHERE (U0."lft" <= 1919
                       AND U0."lft" >= 2
                       AND U0."tree_id" = 23))
             AND NOT ("products_sqlproduct"."is_archived" = TRUE))
```